docs/book/v5/tutorials/create-book-module.md

```diff
@@ -542,8 +542,8 @@ class BookHandler extends AbstractHandler implements RequestHandlerInterface
         "config"
     )]
     public function __construct(
-        protected HalResponseFactory $responseFactory,
-        protected ResourceGenerator $resourceGenerator,
+        protected ?HalResponseFactory $responseFactory,
+        protected ?ResourceGenerator $resourceGenerator,
```

Comment:
```suggestion
        protected ?ResourceGenerator $resourceGenerator = null,
```

Reply:
@alexmerlin why do you suggest default value to null also?

Reply:
@zio-mitch Because the types of these properties indicate that they can be either `null` or the specified class:
- `?HalResponseFactory` = `null` or `HalResponseFactory`
- `?ResourceGenerator ` = `null` or `ResourceGenerator `

If, for some reason they are not set, then calling their getters will throw an error.

Reply:
OK but for clarity of contract would not be better to have the end user know he is passing null explicitly? @alexmerlin

Reply:
Lat's take [HomeHandler.php](https://github.com/dotkernel/api/blob/5.0/src/App/src/Handler/HomeHandler.php) as an example.
This handler does not need either of those two properties, because it simply returns a JSON response (that's why it's no need to add `HomeHandler::class` in `src/App/src/ConfigProvider.php` under the `delegators` key).
If the two properties were defined without the default `null`, then if you were to call either of them, you would get a property not initialized error.

Hope this clarifies your concerns. If not, please let me know and we can discuss further.